packages/collector/test/tracing/messaging/node-rdkafka/test.js

```diff
@@ -0,0 +1,290 @@
+ * Otherwise, there is no way to append the Instana headers to it.
+ * * The Producer, as stream or standard API cannot propagate span correlation when headerFormat is set to 'binary'.
+ * More info here: https://github.com/Blizzard/node-rdkafka/pull/935.
+ * * If the option dr_cb is not set to true, we cannot guarantee that a message was sent, but a span with a successful
+ * sent message will be created.
+ */
+
+const path = require('path');
+const { expect } = require('chai');
+const { fail } = expect;
+const {
+  tracing: { constants }
+} = require('@instana/core');
+
+const {
+  tracing: { supportedVersion }
+} = require('@instana/core');
+
+const config = require('../../../../../core/test/config');
+const { expectExactlyOneMatching, retry, delay, stringifyItems } = require('../../../../../core/test/test_util');
+const ProcessControls = require('../../../test_util/ProcessControls');
+const globalAgent = require('../../../globalAgent');
+const { verifyHttpRootEntry, verifyHttpExit } = require('@instana/core/test/test_util/common_verifications');
+
+let mochaSuiteFn;
+
+const producerEnableDeliveryCbOptions = ['true', 'false'];
+const producerApiMethods = ['standard', 'stream'];
+const consumerApiMethods = ['standard', 'stream'];
+
+const { getCircularList } = require('@instana/core/test/test_util/circular_list');
+const getNextProducerMethod = getCircularList(producerApiMethods);
+const getNextConsumerMethod = getCircularList(consumerApiMethods);
+const getNextDeliveryCb = getCircularList(producerEnableDeliveryCbOptions);
+
+if (!supportedVersion(process.versions.node)) {
+  mochaSuiteFn = describe.skip;
+} else {
+  mochaSuiteFn = describe;
+}
+const retryTime = config.getTestTimeout() * 2;
+
+const topic = 'rdkafka-topic';
+
+mochaSuiteFn('tracing/messaging/node-rdkafka', function () {
+  this.timeout(config.getTestTimeout() * 4);
+
+  globalAgent.setUpCleanUpHooks();
+  const agentControls = globalAgent.instance;
+
+  describe('tracing enabled, no suppression', function () {
+    const producerControls = new ProcessControls({
+      appPath: path.join(__dirname, 'producer'),
+      port: 3216,
+      useGlobalAgent: true,
+      env: {
+        RDKAFKA_PRODUCER_DELIVERY_CB: getNextDeliveryCb()
+      }
+    });
+
+    ProcessControls.setUpHooksWithRetryTime(retryTime, producerControls);
+
+    consumerApiMethods.forEach(consumerMethod => {
+      describe(`consuming via ${consumerMethod} API`, () => {
+        const consumerControls = new ProcessControls({
+          appPath: path.join(__dirname, 'consumer'),
+          port: 3215,
+          useGlobalAgent: true,
+          env: {
+            RDKAFKA_CONSUMER_AS_STREAM: consumerMethod === 'stream' ? 'true' : 'false'
+          }
+        });
+
+        ProcessControls.setUpHooksWithRetryTime(retryTime, consumerControls);
+
+        producerApiMethods.forEach(producerMethod => {
+          [false, 'sender'].forEach(withError => {
```

Comment:
Is it possible that something is missing in the tests regarding error cases?

See line https://github.com/instana/nodejs/pull/486/files#diff-1b77510f78eb55910153dd198bee01c7d0e63aa32d30da88adae14562db0d5f8R143

`withError` is never `receiver` as far as I can see.

Furthermore, there are three cases where we set `span.ec` in the instrumentation, but it looks like only [one case](https://github.com/instana/nodejs/pull/486/files#diff-24bbb2057190b9840c5c35b03ff3849adf0501722464d3ba476378bf0dfd8fd4R162) is covered.

Reply:
> withError is never receiver as far as I can see

The reason why `receiver` is never set is because I could not find a way to force an error when receiving a message. Yet I kept the option in the test in case we find ways to reproduce errors in the future.

>  there are three cases where we set span.ec in the instrumentation

The uncovered cases include either an event emit from deliveryCb or a received message. We can discuss together, but as far as I could investigate, I could not properly force/capture errors in these cases

Reply:
Yeah I totally understand 

I might try to increase code coverage before we merge the PR if that is okay for you :)